doc/nixbuild.net.md

```diff
@@ -0,0 +1,96 @@
+
+Instead of an entry like:
+
+```
+Host eu.nixbuild.net
+  PubkeyAcceptedKeyTypes ssh-ed25519
+  IdentityFile /path/to/your/private/key
+```
+
+Your entry should look like:
+
+```
+# when connecting to eu.nixbuild.net
+Host eu.nixbuild.net
+  # don't use any authentication mechanisms SSH expects (password, SSH keys, etc.)
+  PreferredAuthentications none
+  # log in as the 'authtoken' user
+  # The token itself contains user information that nixbuild.net needs
+  User authtoken
+  # Send the 'token' environment variable from Nix to nixbuild.net in the SSH session.
+  SendEnv token
+  # Alternatively to SendEnv, you can have a line like:
+  #   SetEnv token=<token>
+  # This will mean you don't have to modify Nix's environment (see below).
+  # But this will involve putting your token, which is a secret, directly into
+  # a configuration file.
+```
+
+and then you should ensure that the environment of your Nix daemon (or, if you're running in single-user mode, your Nix build command) has the `token` variable set to the contents of your token.
+
+Managing access tokens
+----------------------
+We are using [Biscuit](biscuitsec.org/) tokens to access nixbuild.net. Among other things, these tokens allow for *offline attenuation*: If you have a token with a certain access, you can freely generate a new token with more restricted access without going through any central service. In this way, you can securely delegate capabilities; for example, given a token obtained as above, you could generate a new token that expires sooner (or, after future upgrades to nixbuild.net, can only spend a certain number of build hours per week) for use in CI.
+
+**Note that Biscuit tokens are secrets and should be managed as such!**
+
+There is a very rich language for attenuating biscuit tokens, and the nixbuild.net team hopes to provide various needed interfaces for limiting access based on various properties of a given build and account; see [the biscuit docs](https://www.biscuitsec.org/docs/getting-started/introduction/) for more information on the former. Currently, however, the only meaningful limitation on a token is having it expire sooner. Using the [biscuit-cli](https://lib.rs/crates/biscuit-cli)[^nixpkgs], you can achieve this with:
+
+```shell
+$ biscuit attenuate my-token --add-ttl "6 weeks" --block "" > new-token
+```
+
+This will take an existing token in the file `my-token`, adds a requirement that it be used within the next 6 weeks, and stores the resulting token in `new-token`. The `--block ""` is needed to tell `biscuit` that you are not adding any additional policy restrictions beyond the TTL.
```

Comment:
```suggestion
This will take an existing token in the file `my-token`, adds a requirement that it only be used within the next 6 weeks, and stores the resulting token in `new-token`. The `--block ""` is needed to tell `biscuit` that you are not adding any additional policy restrictions beyond the TTL.
```

Reply:
Added and rebased.